ff-net/nn/Layer.js

```diff
@@ -10,6 +10,21 @@ class Layer {
     this.outputNeuronGroup = args.outputNeuronGroup;
   }
 
+  backward(args = {}) {
+    const links = [];
+    
+    this.outputNeuronGroup.neurons.forEach(neuron => {
+      neuron.backward(args);
+      neuron.backLinks.forEach((link) => {
+        links.push(link);
+      });
+    });
+
+    links.forEach(link => {
+      link.backward(args);
+    });
```

Comment:
I guess we could just do this without the intermediate `links` list and run backward on every input link as we visit each output neuron, but it makes more intuitive sense to me to run the backward pass on the output neurons first, and then on the input links.